src/app/record/components/work-stack/work-stack.component.ts

```diff
@@ -0,0 +1,134 @@
+})
+export class WorkStackComponent implements OnInit {
+  @HostBinding('class.display-the-stack') displayTheStackClass = false
+  _workStack: WorkGroup
+  visibility: VisibilityStrings
+  @Input()
+  set workStack(value: WorkGroup) {
+    this._workStack = value
+    this.setInitialStates(value)
+  }
+  get workStack(): WorkGroup {
+    return this._workStack
+  }
+
+  _displayTheStack = false
+  set displayTheStack(mode: boolean) {
+    this._displayTheStack = mode
+    this.displayTheStackClass = this._displayTheStack
+    this.setInitialStates(this.workStack, true)
+  }
+  get displayTheStack(): boolean {
+    return this._displayTheStack
+  }
+
+  // orgDisambiguated: { [key: string]: OrgDisambiguated | null } = {}
+  stackPanelsDisplay: { [key: string]: { topPanelOfTheStack: boolean } } = {}
+  panelDetailsState: {
+    [key: string]: {
+      state: boolean
+    }
+  } = {}
+
+  constructor(
+    private _workService: RecordWorksService,
+    private _organizationsService: OrganizationsService
+  ) {}
+
+  /**
+   * Set the panelDetails and top of the stack card to default mode
+   */
+  private setInitialStates(group: WorkGroup, force = false) {
+    group.works.forEach((work) => {
+      this.setDefaultPanelsDisplay(work, force)
+      this.setDefaultPanelDetailsState(work, force)
+    })
+  }
+
+  /**
+   * On start, hide the details for all the panels
+   */
+  private setDefaultPanelDetailsState(work: Work, force = false) {
+    if (this.panelDetailsState[work.putCode.value] === undefined || force) {
+      this.panelDetailsState[work.putCode.value] = {
+        state: false,
+      }
+    }
+  }
+
+  /**
+   * On start, set the preferred source as the top panel of the stack
+   */
+  private setDefaultPanelsDisplay(work: Work, force = false) {
+    if (this.stackPanelsDisplay[work.putCode.value] === undefined || force) {
+      this.stackPanelsDisplay[work.putCode.value] = {
+        topPanelOfTheStack: this.isPreferred(work) ? true : false,
+      }
+    }
+  }
+
+  isPreferred(work: Work) {
+    const response =
+      work && this.workStack
+        ? this.workStack.defaultPutCode.toString() === work.putCode.value
+        : false
+    return response
+  }
+
+  /**
+   * Show and hide details of the panel
+   */
+  toggleDetails(work: Work) {
+    const putCode = work.putCode.value
+    this.panelDetailsState[putCode].state = !this.panelDetailsState[putCode]
+      .state
+
+    if (this.panelDetailsState[putCode].state) {
+      this.getDetails(work).subscribe((response) => {
+        // this.orgDisambiguated[putCode] = response[0] || null
+      })
+    }
+  }
+
+  /**
+   * Get require extra backend data to display on the panel details
+   */
+  private getDetails(work: Work): Observable<WorksEndpoint> {
+    const putCode = work.putCode.value
+    const $workDetails = this._workService.getDetails(putCode)
+    return $workDetails
+  }
+
+  makePrimaryCard(work: Work) {
+    // TODO
```

Comment:
Missing information in `TODO` and remove console.log

Reply:
I am keeping the TODO since this will be implemented on Q2. 
I have removed the console to avoid showing that on the public page